### PR TITLE
fence_cyberpower_ssh: new fence agent

### PR DESCRIPTION
**Pull request by jwbernin**
This agent was created using fence_apc as a guide. I've tested it on a command line and within oVirt and it behaves correctly for those two cases.

### CODE REVIEW

**Review by oalbrigt — CHANGES_REQUESTED** [review]
I've added some comments for a couple of minor changes.

The directory name should also contain _ssh.

And run `make xml-upload` and add the metadata it generates to the PR (or it'll fail CI when I accept it for testing).

You can also add a subpackage in fence-agents.spec.in for RPM-based distros.

File: fence-agents.spec.in [hunk 1]
```diff
@@ -534,6 +535,21 @@ Fence agent for Nova compute nodes.
 %{_mandir}/man8/fence_evacuate.8*
 %endif
 
+%package cyberpower-ssh
+License: GPLv2+ and LGPLv2+
+Summary: Fence agent for CyberPower network PDUs
+%if 0%{?fedora} || 0%{?centos} > 7 || 0%{?rhel} > 7 || 0%{?suse_version}
```
Comment (oalbrigt) [inline fence-agents.spec.in:541]: You can replace this if/else section with `Requires: openssh-clients`.

**Comment by jwbernin** [comment]
Should I add the fence_cyberpower_ssh.xml file to the git commit for upload, or upload it in a different way?

File: agents/cyberpower/fence_cyberpower_ssh.py [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+
+import sys, re, time
+import atexit
+sys.path.append("@FENCEAGENTSLIBDIR@")
+from fencing import *
+from fencing import fail, fail_usage, EC_STATUS
+
+def set_power_status(conn, options):
+
+	conn.send_eol("oltctrl index "+options["--plug"]+" act delay"+options["--action"])
```
[the remark this answers is not in the document]
Comment (jwbernin) [reply agents/cyberpower/fence_cyberpower_ssh.py]: Change pushed to MR

File: agents/cyberpower/fence_cyberpower_ssh.py [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+	else:
+		try:
+			(_,status) = outlets[options["--plug"]]
+			return status.lower().strip()
+		except KeyError:
+			fail(EC_STATUS)
+
+def main():
+	device_opt = ["ipaddr", "login", "passwd", "cmd_prompt", "secure", \
+			"port", "switch", "telnet"]
```
[the remark this answers is not in the document]
Comment (jwbernin) [reply agents/cyberpower/fence_cyberpower_ssh.py]: Change pushed to MR

File: fence-agents.spec.in (hunk 1 above)
Comment (jwbernin) [reply fence-agents.spec.in]: Change pushed to MR.

**Comment by oalbrigt** [comment]
> Should I add the fence_cyberpower_ssh.xml file to the git commit for upload, or upload it in a different way?

Add it to the git commit.

**Comment by jwbernin** [comment]
I believe I've pushed all the requested changes, including the upload of the `make make-upload` output.

File: agents/cyberpower_ssh/fence_cyberpower_ssh.py [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+	docs["shortdesc"] = "Fence agent for CyberPower over ssh"
+	docs["longdesc"] = "fence_cyberpower_ssh is an I/O Fencing agent \
+which can be used with the CyberPower network power switch. It logs into \
+device via ssh and reboots a specified outlet. Lengthy ssh connections \
+should be avoided while a GFS cluster is running because the connection \
+will block any necessary fencing actions."
+	docs["vendorurl"] = "http://www.cyberpower.com"
+	show_docs(options, docs)
+
+	## Support for --plug [switch]:[plug] notation that was used before
```
Comment (oalbrigt) [inline agents/cyberpower_ssh/fence_cyberpower_ssh.py:60]: We can remove this as we're not using the --switch parameter in this agent (remove it in device_opt as well), and run `make xml-upload` to update the test metadata.

**Comment by oalbrigt** [comment]
ok to test

File: fence-agents.spec.in [hunk 5]
```diff
@@ -534,6 +535,17 @@ Fence agent for Nova compute nodes.
...
+%package cyberpower-ssh
+License: GPLv2+ and LGPLv2+
+Summary: Fence agent for CyberPower network PDUs
+Requires: openssh-clients
+Requires: fence-agents-common = %{version}-%{release}
+BuildArch: noarch
+%description cyberpower-ssh
+%files cyberpower-ssh
+%{_sbindir}/fence_cyberpower_ssh
+%{_mandir}/man8/fence_cyberpower.8*
```
Comment (oalbrigt) [inline fence-agents.spec.in:547]: You forgot _ssh here.

File: agents/cyberpower_ssh/fence_cyberpower_ssh.py [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+#####
+
+import sys, re, time
+import atexit
+sys.path.append("@FENCEAGENTSLIBDIR@")
+from fencing import *
+from fencing import fail, EC_STATUS
+
+def set_power_status(conn, options):
+
```
Comment (oalbrigt) [inline agents/cyberpower_ssh/fence_cyberpower_ssh.py:17]: You should remove this line.

Comment (oalbrigt) [reply agents/cyberpower_ssh/fence_cyberpower_ssh.py:16]: Not the import line... the empty line at the beginning of the set_power_status().

Comment (jwbernin) [reply agents/cyberpower_ssh/fence_cyberpower_ssh.py:16]: Next commit coming soon...

Comment (oalbrigt) [reply agents/cyberpower_ssh/fence_cyberpower_ssh.py]: You'll need to readd the import-line, or it'll fail.

Comment (jwbernin) [reply agents/cyberpower_ssh/fence_cyberpower_ssh.py]: The fail and EC_STATUS items aren't imported by the "from fencing import *" line?

Comment (oalbrigt) [reply agents/cyberpower_ssh/fence_cyberpower_ssh.py]: No. * imports a pre-defined set of the fencing library.

**Comment by oalbrigt** [comment]
LGTM. Thanks.

**Pull request merged**